src/modules/Profile.rb

```diff
@@ -14,6 +14,33 @@
 require "ui/password_dialog"
 
 module Yast
+  class ProfileHash < Hash
+    # replace Hash -> ProfileHash recursively
+    def initialize(x = {})
+      x.each_pair do |key, value|
+        if value.is_a?(Hash)
+          self[key] = ProfileHash.new(value)
+        else
+          self[key] = value
+        end
+      end
+    end
+
+    # with an explicit default it's possible to check for the presence of an
+    # element vs. empty element
+    def fetch_as(x, type, default = nil)
+      tmp = self.fetch(x, nil)
+      if !tmp.is_a?(type)
```

Comment:
I expect logging there as it is still old or invalid profile, so we know that something unexpected is there.